src/input/reader_to_storage.cc

```diff
@@ -632,7 +633,7 @@ StorageBase * ReaderToStorage::make_storage_from_default(const string &dflt_str,
 
     } catch (Input::Type::ExcWrongDefault & e) {
         // message to distinguish exceptions thrown during Default value check at declaration
-        xprintf(Msg, "Wrong default value while reading an input stream:\n");
+    	e << Type::EI_Desc("Wrong default value while reading an input stream:\n");
```

Comment:
Introduce new exception in case of invalid JSON format (in method Default::check_validity).
Here catch both exceptions, formate their what message and make a fatal error (either using assert or throwing an fatel exception directly). Add there what message of the original exception.